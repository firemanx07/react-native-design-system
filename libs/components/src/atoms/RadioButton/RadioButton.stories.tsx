import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';

import { StorybookScreen, StorybookSection } from '../../storybook';
import { default as RadioButtonBase, RadioButtonProps } from './RadioButton';

const Story: ComponentMeta<typeof RadioButtonBase> = {
  component: RadioButtonBase,
  title: 'Atoms/RadioButton',
  argTypes: {
    selected: {
      control: { type: 'select' },
      options: [true, false],
    },
  },
};
export default Story;
const noop = () => {
  // This is a mock
};
const Template: ComponentStory<typeof RadioButtonBase> = args => {
  return (
    <StorybookScreen>
      <StorybookSection title="Radio Button :">
        <RadioButtonBase {...args} />
      </StorybookSection>
      <StorybookSection title="Disabled:">
        <RadioButtonBase {...args} disabled />
      </StorybookSection>
      <StorybookSection title="active:">
        <RadioButtonBase {...args} disabled={false} selected />
      </StorybookSection>
      <StorybookSection title="inactive:">
        <RadioButtonBase {...args} disabled={false} selected={false} />
      </StorybookSection>
    </StorybookScreen>
  );
};

const parameters = {
  controls: {
    include: ['size', 'value', 'selected', 'disabled'],
  },
};

const defaultArgs: RadioButtonProps = {
  value: 'option 1',
  size: 24,
  selected: false,
  disabled: false,
  onSelect: noop,
};

export const RadioButton = Template.bind({});
RadioButton.parameters = parameters;
RadioButton.args = {
  ...defaultArgs,
};
