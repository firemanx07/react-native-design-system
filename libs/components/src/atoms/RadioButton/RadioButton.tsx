import { styled, useNamespacedTheme } from '@proxym/themes';
import { rgba } from 'polished';
import React, { memo } from 'react';
import { StyleProp, ViewStyle } from 'react-native';

import { TestIDType } from '../../types';
import { Circle } from '../Circle';

export type RadioButtonProps = {
  value: string;
  selected: boolean;
  disabled?: boolean;
  size?: number;
  style?: StyleProp<ViewStyle>;
  onSelect: (value: string) => void;
} & TestIDType;

const BORDER_WIDTH = 2;
const RadioButton = ({
  value,
  selected,
  onSelect,
  size = 24,
  disabled,
  testID,
}: RadioButtonProps) => {
  const { colors, spacing, iconSize } = useNamespacedTheme();
  const INNER_CIRCLE_SIZE =
    size - (spacing.primarySmall * size) / iconSize.primary;
  const disabledStyle: ViewStyle = {
    borderColor: rgba(colors.grayDark, 0.1),
    backgroundColor: rgba(colors.grayDark, 0.3),
  };

  return (
    <RadioButtonContainer
      size={size - BORDER_WIDTH}
      selected={selected}
      onPress={() => onSelect(value)}
      style={disabled && disabledStyle}
      testID={testID}
    >
      {!disabled ? (
        selected && <Circle color={colors.secondary} size={INNER_CIRCLE_SIZE} />
      ) : (
        <Circle color={rgba(colors.grayDark, 0.5)} size={INNER_CIRCLE_SIZE} />
      )}
    </RadioButtonContainer>
  );
};

export default memo(RadioButton);

type RadioButtonContainerProps = {
  size: number;
  disabled?: boolean;
  selected?: boolean;
};
const RadioButtonContainer = styled.TouchableOpacity<RadioButtonContainerProps>`
  border-radius: ${({ size }) => size / 2};
  width: ${({ size }) => size}px;
  height: ${({ size }) => size}px;
  border-width: ${BORDER_WIDTH}px;
  border-color: ${({ theme, selected }) =>
    !selected ? theme.ds.colors.grayDark : theme.ds.colors.secondary};
  align-items: center;
  justify-content: center;
`;
