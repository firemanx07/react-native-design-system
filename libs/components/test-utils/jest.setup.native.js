import { Animated } from 'react-native';

require('react-native-reanimated/lib/reanimated2/jestUtils').setUpTests();
require('react-native-gesture-handler/jestSetup.js');

console.warn = () => {
  //this is mock
};

Animated['timing'] = () => ({
  start: () => jest.fn(),
});
