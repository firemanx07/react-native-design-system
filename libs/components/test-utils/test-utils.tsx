import { theme, ThemeProvider } from '@celebrate-app/themes';
import { Story } from '@storybook/react';
import {
  cleanup,
  render,
  RenderOptions,
  waitFor,
} from '@testing-library/react-native';
import React, { ReactNode } from 'react';

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions
) => {
  return render(ui, { wrapper: Providers, ...options });
};

beforeEach(cleanup);
afterEach(cleanup);

export function testStory(displayName: string, TestComponent: Story) {
  test(`${displayName} snapshot test`, async () => {
    const comp = customRender(<TestComponent {...TestComponent.args} />);
    jest.runOnlyPendingTimers();

    await waitFor(() => {
      expect(comp).toBeTruthy();
      expect(comp.toJSON()).toMatchSnapshot();
    });
  });
}

export function testComponentSnapshot(
  TestComponent: React.ReactElement<
    any,
    string | React.JSXElementConstructor<any>
  >
) {
  test('Snapshot test', () => {
    const comp = customRender(TestComponent);
    expect(comp).toBeTruthy();
    expect(comp.toJSON()).toMatchSnapshot();
  });
}

export function testAllStories(stories: any) {
  Object.keys(stories).forEach((story) => {
    if (story !== 'default') {
      testStory(story, stories[story]);
    }
  });
}
