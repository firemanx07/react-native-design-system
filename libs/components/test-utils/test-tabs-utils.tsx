import { waitFor } from '@testing-library/react-native';

import { customRender } from './test-utils';

type renderTabsProps = {
  elem: JSX.Element;
  testIDs: string[];
};
export const renderTabs = ({ elem, testIDs }: renderTabsProps) => {
  const comp = customRender(elem);
  const tabs: ReturnType<typeof comp.getByTestId>[] = [];
  testIDs.forEach(testID => tabs.push(comp.getByTestId(testID)));
  return { comp, tabs };
};
export const testTabs = async (args: ReturnType<typeof renderTabs>) => {
  await waitFor(() => {
    expect(args.comp).toBeTruthy();
    args.tabs.forEach(tab => expect(tab).toBeDefined());
  });
};
