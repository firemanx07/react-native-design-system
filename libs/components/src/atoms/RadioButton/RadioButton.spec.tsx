import { testAllStories } from '../../../test-utils';
import * as stories from './RadioButton.stories';

describe('RadioButton tests', () => {
  testAllStories(stories);
});
