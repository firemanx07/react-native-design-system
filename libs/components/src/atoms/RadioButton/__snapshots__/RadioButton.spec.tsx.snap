// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton tests RadioButton snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minHeight": 600,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "borderColor": "#2C2C2C",
          "borderRadius": 4,
          "borderStyle": "dashed",
          "borderWidth": 1,
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 370,
        },
      ]
    }
    width={370}
  >
    <Text
      fontFamily="Rubik-Bold"
      fontSize={16}
      lineHeight={20}
      style={
        [
          {
            "color": "#2C2C2C",
            "fontFamily": "Rubik-Bold",
            "fontSize": 16,
            "lineHeight": 20,
          },
          {
            "backgroundColor": "#C7C7D2",
            "left": 16,
            "paddingHorizontal": 8,
            "position": "absolute",
            "top": -12,
          },
        ]
      }
    >
      Radio Button :
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#C7C7D2",
          "borderRadius": 11,
          "borderWidth": 2,
          "height": 22,
          "justifyContent": "center",
          "opacity": 1,
          "width": 22,
        }
      }
    />
  </View>
  <View
    style={
      [
        {
          "borderColor": "#2C2C2C",
          "borderRadius": 4,
          "borderStyle": "dashed",
          "borderWidth": 1,
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 370,
        },
      ]
    }
    width={370}
  >
    <Text
      fontFamily="Rubik-Bold"
      fontSize={16}
      lineHeight={20}
      style={
        [
          {
            "color": "#2C2C2C",
            "fontFamily": "Rubik-Bold",
            "fontSize": 16,
            "lineHeight": 20,
          },
          {
            "backgroundColor": "#C7C7D2",
            "left": 16,
            "paddingHorizontal": 8,
            "position": "absolute",
            "top": -12,
          },
        ]
      }
    >
      Disabled:
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(199,199,210,0.3)",
          "borderColor": "rgba(199,199,210,0.1)",
          "borderRadius": 11,
          "borderWidth": 2,
          "height": 22,
          "justifyContent": "center",
          "opacity": 1,
          "width": 22,
        }
      }
    >
      <View
        color="rgba(199,199,210,0.5)"
        size={12}
        style={
          [
            {
              "backgroundColor": "rgba(199,199,210,0.5)",
              "borderRadius": 6,
              "height": 12,
              "width": 12,
            },
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      [
        {
          "borderColor": "#2C2C2C",
          "borderRadius": 4,
          "borderStyle": "dashed",
          "borderWidth": 1,
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 370,
        },
      ]
    }
    width={370}
  >
    <Text
      fontFamily="Rubik-Bold"
      fontSize={16}
      lineHeight={20}
      style={
        [
          {
            "color": "#2C2C2C",
            "fontFamily": "Rubik-Bold",
            "fontSize": 16,
            "lineHeight": 20,
          },
          {
            "backgroundColor": "#C7C7D2",
            "left": 16,
            "paddingHorizontal": 8,
            "position": "absolute",
            "top": -12,
          },
        ]
      }
    >
      active:
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#49C4E9",
          "borderRadius": 11,
          "borderWidth": 2,
          "height": 22,
          "justifyContent": "center",
          "opacity": 1,
          "width": 22,
        }
      }
    >
      <View
        color="#49C4E9"
        size={12}
        style={
          [
            {
              "backgroundColor": "#49C4E9",
              "borderRadius": 6,
              "height": 12,
              "width": 12,
            },
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      [
        {
          "borderColor": "#2C2C2C",
          "borderRadius": 4,
          "borderStyle": "dashed",
          "borderWidth": 1,
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 370,
        },
      ]
    }
    width={370}
  >
    <Text
      fontFamily="Rubik-Bold"
      fontSize={16}
      lineHeight={20}
      style={
        [
          {
            "color": "#2C2C2C",
            "fontFamily": "Rubik-Bold",
            "fontSize": 16,
            "lineHeight": 20,
          },
          {
            "backgroundColor": "#C7C7D2",
            "left": 16,
            "paddingHorizontal": 8,
            "position": "absolute",
            "top": -12,
          },
        ]
      }
    >
      inactive:
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#C7C7D2",
          "borderRadius": 11,
          "borderWidth": 2,
          "height": 22,
          "justifyContent": "center",
          "opacity": 1,
          "width": 22,
        }
      }
    />
  </View>
</View>
`;
